packages/edit-site/src/components/block-editor/use-post-link-props.js

```diff
@@ -15,6 +15,10 @@ export function usePostLinkProps() {
 	const history = useHistory();
 
 	return ( params, state ) => {
-		return getPostLinkProps( history, params, state );
+		return getPostLinkProps(
+			history,
+			{ ...params, focusMode: true },
+			state
+		);
```

Comment:
I know that this was introduced previously, but there's a few things that are wrong with this hook being passed as is to `EditorProvider` settings. The main thing is that this hook returns a generic function that is returned to navigate in the site editor. It also assumes a special kind of "url" in the site editor . In other words, It's not really clear that it's a `{ postType, postId } => goTo( someUrlBasedOnPostTypePostId )` so tomorrow if the site editor changes its routes and routing, it will break directly and we won't notice.

I also noticed that in places where there should be no awareness of site editor (like the `block/edit.js` file of the reusable block, we're passing `canvas: 'edit',` to this function because of the weird behavior above. This is bad because a block shouldn't know that it's within a site editor and that there's something called "canvas". In other words, this function should only have "postId" and "postType" as arguments and nothing more.

Reply:
Also it seems the site editor is not providing the `goBack` no? Is that missing?

Reply:
The thread about goBack in the site editor [is here](https://github.com/WordPress/gutenberg/pull/57700#discussion_r1462687050). I have removed the `canvas: 'edit'` param from all locations outside the site editor.

Reply:
>   The main thing is that this hook returns a generic function that is returned to navigate in the site editor. It also assumes a special kind of "url" in the site editor . In other words, It's not really clear that it's a { postType, postId } => goTo( someUrlBasedOnPostTypePostId ) so tomorrow if the site editor changes its routes and routing, it will break directly and we won't notice.

It is using [the same](https://github.com/WordPress/gutenberg/blob/refactor/template-only-mode-in-post-editor/packages/edit-site/src/components/routes/link.js#L24) `history.push( params, state );` call as the other navigations in the site editor though, so in theory it is not assuming anything about the site editor url format, it is just assuming that if it passes a `postId` and a `postType` to the site editor then the site editor's router will know how to navigate to it. 

Now that the `canvas: edit` param has been pulled back into the site editor (thanks for picking up on that issue) if the site editor url format changes and this param is no longer valid then I assume this settings usage of it will be updated to the new format during the search and replace of all the other `canvas: edit` router param settings in the site editor package.

I may be misunderstanding your concerns here though.